### ISSUE TITLE
calendar displaying incorrect dates

### ISSUE THREAD

**Issue by benrlodge** [Hacker0x01/react-datepicker#160]
Great component thanks, just noticed an issue though. As seen below, July 25, 2015 should be a Saturday, not a Sunday. Sometimes I'll see it correct, but not always. FWIW the date that is saved is correct, just what's displaying on the calendar. 

![screen shot 2015-08-11 at 9 25 56 pm](https://cloud.githubusercontent.com/assets/3275453/9214762/c179c554-406f-11e5-89bf-e2eec749b1c3.png)

![screen shot 2015-08-11 at 9 32 17 pm](https://cloud.githubusercontent.com/assets/3275453/9214814/7fe443b6-4070-11e5-8044-af7359ca0510.png)

**Comment by blackjk3** [comment]
We are also experiencing the same behavior.  @benrlodge were you thinking about submitting a PR? Otherwise, we could look into it at well.

**Comment by benrlodge** [comment]
@blackjk3 this was just for a personal project so I haven't revisited it, likely won't in the future.

**Comment by GalacticWeb** [comment]
+1 !!!

**Comment by gl-kunalgupta** [comment]
+1

**Comment by martijnrusschen** [comment]
Did someone track down the cause of this issue? Would love to fix this.

**Comment by ge3kusa** [comment]
We use this in a few spots and one instance renders the dates correctly. The other wouldn't until I forced the start of the week to be Sunday by passing weekStart="0" as a param. It cleared up after that. I'm not calling that a solution but it might lead to one in the future. At the very least it got us past the issue.

**Comment by dnl** [comment]
I used this as a hacky workaround.

```
  // moment.weekdaysMin is [sun, mon, tue...]
  // DatePicker.weekdays needs [sat, sun, mon...] if weekStart=1
  //                           [sun, mon, tue...] if weekStart=0
  // if this isn't used it will label the first column 'tuesday' for 'starts on monday' locale.
  // DatePicker appears to be doing the offset twice.
  weekDays () {
    var offset = moment.localeData().firstDayOfWeek();
    var weekdays = moment.weekdaysMin();
    if (offset > 0) {
      var start = weekdays.slice(7-offset);
      var end = weekdays.slice(0,-offset);
      return end.concat(start);
    } else {
      return weekdays;
    }
  }
```

**Comment by rafeememon** [comment]
Probably fixed by #277
